lib/charms/operator_libs_linux/v0/systemd.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Abstractions for stopping, starting and managing system services via systemd.
+
+This library assumes that your charm is running on a platform that uses systemd. E.g.,
+Centos 7 or later, Ubuntu Xenial (16.04) or later.
+
+For the most part, we transparently provide an interface to a commonly used selection of
+systemd commands, with a few shortcuts baked in. For example, service_pause and
+service_resume with run the mask/unmask and enable/disable invocations.
+
+Example usage:
+```python
+from systemd import *
```

Comment:
I wonder if just `import systemd` is better? Feels like that + some slight renaming might be more natural? `systemd.start(service)`, `systemd.stop(service)`? I could go either way though.

Reply:
I went back and forth on this. systemd.stop, systemd.start etc is pretty natural. But so is service_start, service_stop, etc. The latter more closely reflects the name of the systemd commands, and also does not imply that we are starting or stopping systemd itself.

I do realize that I tend to be enthusiastically wordy in my variable names (perhaps weirdly, for someone who uses a bare text editor with no autocomplete).

I'll give it some more thought ...

Reply:
Yeh, I'm not super hard over either way. Was thinking aloud :)